packages/rrweb-snapshot/src/snapshot.ts

```diff
@@ -194,23 +194,45 @@ function getAbsoluteSrcsetString(doc: Document, attributeValue: string) {
   return output.join(', ');
 }
 
+const cachedDocument = new WeakMap<Document, HTMLAnchorElement>();
+
 export function absoluteToDoc(doc: Document, attributeValue: string): string {
   if (!attributeValue || attributeValue.trim() === '') {
     return attributeValue;
   }
-  const a: HTMLAnchorElement = doc.createElement('a');
-  a.href = attributeValue;
-  return a.href;
+
+  return getHref(doc, attributeValue);
```

Comment:
> @H4ad Could you also test the following possibility:
> 
> ```
> function getHref(doc: Document) {
>   // return a href without hash
>   const url = new URL(doc.location.href);
>   url.hash = ''
>   return url.href;
> }
> ```
> 
> Some more work would need to be done to support the new `customHref` case so maybe this PR is fine

A slight change based on @eoghanmurray 's suggestion
```suggestion

  const url = new Url(attributeValue, doc.location);
  url.hash = '';
  return url.href;
```